statsmodels/tsa/statespace/kalman_filter.py

```diff
@@ -1271,3 +1576,149 @@ def _predict(self, nstatic, ndynamic, nforecast, model):
         results = FilterResults(model)
         results.update_filter(kfilter)
         return results
+
+
+class PredictionResults(FilterResults):
```

Comment:
This class is a different kind of wrapper; it only wraps `FilterResults` to restrict it to the appropriate time range. The `__getattr__` is there for lazy-restriction of the various results. It is somewhat "separate" from the Statsmodels infrastructure, in the same way that `FilterResults` is separate. 

The parallel is this: `mlemodel.MLEResults` delegates to `mlemodel.FilterResults`, and `mlemodel.PredictionResults` delegates to `kalman_filter.PredictionResults`.

Then `mlemodel.MLEResultsWrapper` and `mlemodel.PredictionResultsWrapper` provide Statsmodels wrapping functionality.